Helpers/opent2t-translator-helper-wink/js/index.js

```diff
@@ -9,179 +9,184 @@ var apiEndpoint = 'https://api.wink.com';
+    getDesiredStateAsync(deviceType, deviceId, field) {
 
-            // q will help us with returning a promise
-            var deferred = q.defer();
+        // q will help us with returning a promise
+        var deferred = q.defer();
 
-            getDeviceDetails(deviceType, deviceId).then((details) => {
+        this.getDeviceDetailsAsync(deviceType, deviceId).then((response) => {
 
-                // successfully got device details, parse out the desired state
-                // of the requested field
-                var data = details.data;
-                if (!!data) {
-                    var desiredStateCollection = data['desired_state'];
+            // successfully got device details, parse out the desired state
+            // of the requested field
+            var data = response.data;
+            if (!!data) {
+                var desiredStateCollection = data['desired_state'];
 
-                    if (!!desiredStateCollection) {
-                        deferred.resolve(desiredStateCollection[field]);
-                    } else {
-                        deferred.reject('Invalid response from server: no desired state collection.');
-                    }
+                if (!!desiredStateCollection) {
+                    deferred.resolve(desiredStateCollection[field]);
                 } else {
-                    deferred.reject('Invalid response from server: no data element.');
+                    deferred.reject('Invalid response from server: no desired state collection.');
                 }
-            }).catch((error) => {
-                // there was an error
-                deferred.reject(error);
-            });
-
-            return deferred.promise;
-        },
+            } else {
+                deferred.reject('Invalid response from server: no data element.');
+            }
+        }).catch((error) => {
+            // there was an error
+            deferred.reject(error);
+        });
+
+        return deferred.promise;
+    }
 
-        getLastReading: function (deviceType, deviceId, field) {
+    // Gets the last reading (of a single field)
+    getLastReadingAsync(deviceType, deviceId, field) {
 
-            // q will help us with returning a promise
-            var deferred = q.defer();
+        // q will help us with returning a promise
+        var deferred = q.defer();
 
-            getDeviceDetails(deviceType, deviceId).then((details) => {
+        this.getDeviceDetailsAsync(deviceType, deviceId).then((details) => {
 
-                // successfully got device details, parse out the last reading
-                // of the requested field
-                var data = details.data;
-                if (!!data) {
-                    var lastReadingsCollection = data['last_reading'];
+            // successfully got device details, parse out the last reading
+            // of the requested field
+            var data = details.data;
+            if (!!data) {
```

Comment:
I'll do this later... for now I like the readability and it isn't a ton of code dupe with the latest rev.